runtime/vm/ClassInitialization.cpp

```diff
@@ -386,6 +386,19 @@ classInitStateMachine(J9VMThread *currentThread, J9Class *clazz, J9ClassInitStat
 				if (NULL == initializationLock) {
 					goto done;
 				}
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+				/* Preparation is the earliest point where the defaultValue would needed. 
+				 * I.e pre-filling static fields. Therefore, the defaultValue must be allocated at 
+				 * the end of verification 
+				 */
+				j9object_t defaultValue = vm->memoryManagerFunctions->J9AllocateObject(currentThread, clazz, J9_GC_ALLOCATE_OBJECT_TENURED | J9_GC_ALLOCATE_OBJECT_NON_INSTRUMENTABLE);
+				if (NULL == defaultValue) {
+					setHeapOutOfMemoryError(currentThread);
+					goto done;
+				}
+				clazz->flattenedClassCache[0].clazz = (J9Class *)defaultValue;
+				clazz = VM_VMHelpers::currentClass(clazz);
+#endif
```

Comment:
`clazz = VM_VMHelpers::currentClass(clazz);` should be done after popping the initLock.

Reply:
Do I also need to do this?
if (VM_VMHelpers::exceptionPending(currentThread)) {
					initializationLock = setInitStatus(currentThread, clazz, J9ClassInitUnverified, initializationLock);
					clazz = VM_VMHelpers::currentClass(clazz);
					goto done;
				}

Reply:
Should I also add the following code after this?
if (VM_VMHelpers::exceptionPending(currentThread)) {
						initializationLock = setInitStatus(currentThread, clazz, J9ClassInitUnverified, initializationLock);
						clazz = VM_VMHelpers::currentClass(clazz);
						goto done;
					}